packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -232,6 +250,112 @@ export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFo
     }
   };
 
+  /**
+   * Callback hooked to the root element. When we receive a blur, we want
+   * to detect if the blur is due to an element being removed from the Zone.
+   * This is done by evaluating if the destination (in ev.relatedTarget) is
+   * null. If so, we need to move the focus to either the next element, or
+   * possibly, the previous element if there is no "next".
+   *
+   * In the case there are no more focusable elements in the zone, we "park"
+   * focus on the root temporarily. If the user tabs elsewhere, we remove
+   * focusablilty from the root. After a zone re-render, if focus is still
+   * "parked", we try to move focus back to the first focusable child.
+   *
+   * Note: Attempting to move this to a React event, rather than using a
+   * native event, broke the implementation.
+   */
+  private _onBlurCapture(ev: FocusEvent) {
+    const { current: root } = this._root;
+    const { target: from } = ev;
+    const isFocusDisappearing = ev.relatedTarget === null || ev.relatedTarget === document;
+
+    // If we are tabbing from parked state, reset parked.
+    if (from === root) {
+      this._setParkedFocus(false);
```

Comment:
I'm curious if this could "blink" the `tabIndex` attribute if parked focus is again set to true in the async call below and if that could cause any unintended side effects. Could this instead be called in the various cases in async or if `!isFocusDisappearing`?

Reply:
i've locked it to call once per mutation using a flag.